src/agent/onefuzz-agent/src/managed/cmd.rs

```diff
@@ -23,6 +43,43 @@ pub async fn run(args: &clap::ArgMatches<'_>) -> Result<()> {
     result
 }
 
+// Periodically check available system memory.
+//
+// If available memory drops below the minimum, exit informatively.
+//
+// Parameterized to enable future configuration by VMSS.
+#[cfg(not(target_os = "macos"))]
+async fn out_of_memory(min_bytes: u64) -> Result<OutOfMemory> {
+    loop {
+        match onefuzz::memory::available_bytes() {
+            Ok(available_bytes) => {
+                if available_bytes < min_bytes {
+                    return Ok(OutOfMemory {
+                        available_bytes,
+                        min_bytes,
+                    });
+                }
+            }
+            Err(err) => {
+                warn!("error querying system memory usage: {}", err);
+                return Err(err);
+            }
+        }
```

Comment:
This seems a bit agressive to stop everything as soon as we are unable to read the memory.
I would suggest raising the error if we fail to read the memory after a small number of attempts

Reply:
For the record, the current behavior only stops the memory watchdog async task. But agreed, we should only give up the memory check after several failures in a row.